openlibrary/core/models.py

```diff
@@ -293,8 +293,27 @@ def get_waitinglist(self):
         """Returns list of records for all users currently waiting for this book."""
         return waitinglist.get_waitinglist_for_book(self.key)
 
-    def get_realtime_availability(self):
-        return lending.get_realtime_availability_of_ocaid(self.get('ocaid'))
+    def inject_ia_metadata(self):
```

Comment:
This feels like an anitpattern; you're extending the underlying model with additional fields, _some of the time_. Someone who sees these fields being used will assume they're things that are always present on `Edition`, and not realize they have to call the state-modifying function `inject_ia_metadata` before using them. (Type wise, you're _changing_ the type). The metadata should be an adjacent object that can be queried; something like:

```python
page.ia_metadata.donor
page.ia_metadata.donor_msg
page.ia_metadata.availability
```

You can create a class to wrap around the metadata, and then have a lazy property on the edition for `ia_metadata` (which is _always_ present).

Reply:
Done